src/main/java/com/derongan/minecraft/deeperworld/extensions/DeeperWorldExtensions.kt

```diff
@@ -0,0 +1,60 @@
+package com.derongan.minecraft.deeperworld.extensions
+
+import org.bukkit.Material
+import org.bukkit.entity.Entity
+import org.bukkit.entity.Item
+import org.bukkit.entity.LivingEntity
+import org.bukkit.entity.Player
+
+internal fun Entity.getVehicleRecursive(): Entity? {
+    var currentVehicle = vehicle
+
+    while (currentVehicle?.vehicle != null) {
+        currentVehicle = currentVehicle.vehicle
+    }
+
+    return currentVehicle
+}
+
+internal fun Entity.getPlayerPassengersRecursive(): List<Player>? {
+    return mutableListOf<Player>().let { playerList ->
```

Comment:
Aren't passengers already a complete list, or are there issues with just getting passengers and filtering by player instances?

Reply:
It might make more sense to change this function into returning all passengers, and then in my actual use-case filtering by player instances yeah. This might also be more widely applicable. I'll change it